config/routes.rb

```diff
@@ -9,6 +9,12 @@
   get "/users/:id", to: "users#show", as: "user"
   
   get "/users/:id/posts", to: "posts#index", as: "user_posts"
+  get "/users/:id/posts/new", to: "posts#new", as: "user_posts_new"
+  post "/users/:id/posts", to: "posts#create", as: "user_posts_create"
+  get "/users/:id/posts/:id/new", to: "comments#new", as: "user_post_comments_new"
+  post "/users/:id/posts/:id", to: "comments#create", as: "user_post_comments_create"
+  post "/users/:id/posts/:id/like", to: "likes#create", as: "user_post_likes"
   get "/users/:id/posts/:id", to: "posts#show", as: "user_post"
```

Comment:
- [ ] It is not a good practice to add `/user/:id` always in any routes , kindly remove it.